dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/program/EventProgramEnrollmentServiceTest.java

```diff
@@ -0,0 +1,121 @@
+import org.hisp.dhis.organisationunit.OrganisationUnitService;
+import org.hisp.dhis.test.integration.PostgresIntegrationTestBase;
+import org.hisp.dhis.trackedentity.TrackedEntity;
+import org.hisp.dhis.trackedentity.TrackedEntityService;
+import org.hisp.dhis.user.User;
+import org.hisp.dhis.user.sharing.Sharing;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.annotation.Transactional;
+
+@Transactional
+class EventProgramEnrollmentServiceTest extends PostgresIntegrationTestBase {
+
+  @Autowired private IdentifiableObjectManager manager;
+
+  @Autowired private EventProgramEnrollmentService eventProgramEnrollmentService;
+
+  @Autowired private ProgramService programService;
+
+  @Autowired private OrganisationUnitService organisationUnitService;
+
+  @Autowired private TrackedEntityService trackedEntityService;
+
+  private Program programA;
+
+  private Program programB;
+
+  private Enrollment enrollmentA;
+
+  private Enrollment enrollmentB;
+
+  private Enrollment enrollmentC;
+
+  @BeforeEach
+  void setUp() {
+    OrganisationUnit organisationUnitA = createOrganisationUnit('A');
+    organisationUnitService.addOrganisationUnit(organisationUnitA);
+    OrganisationUnit organisationUnitB = createOrganisationUnit('B');
+    organisationUnitService.addOrganisationUnit(organisationUnitB);
+
+    TrackedEntity trackedEntityA = createTrackedEntity(organisationUnitA);
+    trackedEntityService.addTrackedEntity(trackedEntityA);
+    TrackedEntity trackedEntityB = createTrackedEntity(organisationUnitB);
+    trackedEntityService.addTrackedEntity(trackedEntityB);
+
+    programA = createProgram('A', new HashSet<>(), organisationUnitA);
+    programService.addProgram(programA);
+    programA.setSharing(Sharing.builder().publicAccess("rwrw----").build());
+    programService.updateProgram(programA);
+
+    programB = createProgram('B', new HashSet<>(), organisationUnitA);
+    programService.addProgram(programB);
+
+    enrollmentA = createEnrollment(programA, trackedEntityA, organisationUnitA);
+    manager.save(enrollmentA);
+
+    enrollmentB = createEnrollment(programB, trackedEntityA, organisationUnitB);
+    manager.save(enrollmentB);
+
+    enrollmentC = createEnrollment(programA, trackedEntityA, organisationUnitB);
+    manager.save(enrollmentC);
+
+    User user =
+        createAndAddUser(
+            false, "user", Set.of(organisationUnitA), Set.of(organisationUnitA), "F_EXPORT_DATA");
+    user.setTeiSearchOrganisationUnits(Set.of(organisationUnitA, organisationUnitB));
+    user.setOrganisationUnits(Set.of(organisationUnitA));
+
+    injectSecurityContextUser(user);
+  }
+
+  @Test
+  void testGetEnrollmentsByProgram() {
+    List<Enrollment> enrollments = eventProgramEnrollmentService.getEnrollments(programA);
+    assertEquals(2, enrollments.size());
```

Comment:
please use `assertContainsOnly` or `assertEquals(List.of(), enrollments)` if order matters

this style of assertions are a pain if they fail on CI which is why we should stop writing them this way 😬